frontend/rust-lib/flowy-database2/src/event_handler.rs

```diff
@@ -920,3 +920,76 @@ pub(crate) async fn update_field_settings_handler(
     .await?;
   Ok(())
 }
+
+pub(crate) async fn get_related_database_ids_handler(
+  _data: AFPluginData<DatabaseViewIdPB>,
+  _manager: AFPluginState<Weak<DatabaseManager>>,
+) -> FlowyResult<()> {
+  Ok(())
+}
+
+#[tracing::instrument(level = "debug", skip_all, err)]
+pub(crate) async fn update_relation_cell_handler(
+  data: AFPluginData<RelationCellChangesetPB>,
+  manager: AFPluginState<Weak<DatabaseManager>>,
+) -> FlowyResult<()> {
+  let manager = upgrade_manager(manager)?;
+  let params: RelationCellChangesetPB = data.into_inner();
+  let view_id = parser::NotEmptyStr::parse(params.view_id)
+    .map_err(|_| flowy_error::ErrorCode::DatabaseViewIdIsEmpty)?
+    .0;
+  let cell_id: CellIdParams = params.cell_id.try_into()?;
+  let params = RelationCellChangeset {
+    inserted_row_ids: params
+      .inserted_row_ids
+      .into_iter()
+      .map(Into::into)
+      .collect(),
+    removed_row_ids: params.removed_row_ids.into_iter().map(Into::into).collect(),
+  };
+
+  let database_editor = manager.get_database_with_view_id(&view_id).await?;
+
+  // // get the related database
+  // let related_database_id = database_editor
+  //   .get_related_database_id(&cell_id.field_id)
+  //   .await?;
+  // let related_database_editor = manager.get_database(&related_database_id).await?;
+
+  // // validate the changeset contents
+  // related_database_editor
+  //   .validate_row_ids_exist(&params)
+  //   .await?;
+
+  // update the cell in the database
+  database_editor
+    .update_cell_with_changeset(&view_id, cell_id.row_id, &cell_id.field_id, params)
+    .await?;
+  Ok(())
+}
+
+pub(crate) async fn get_related_row_datas_handler(
```

Comment:
Data is plural 👀

Reply:
Yeah... I'm thinking row data as a struct, so plural of that xD